### PR TITLE
allow coverage to be not there

### PR DESCRIPTION
**Pull request by uwej711**
The quality gate decoration fails with an exception when sonarqube has no coverage information. I think it is better to assume coverage "0" than to fail.

### CODE REVIEW

**Comment by kortov** [comment]
relates to #85

**Comment by uwej711** [comment]
Didn't see the other pr, this is working for Sonarqube 7.9.2

File: src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabServerPullRequestDecorator.java [hunk 1]
```diff
@@ -146,11 +146,11 @@ public void decorateQualityGateStatus(AnalysisDetails analysis, UnifyConfigurati
...
-            QualityGate.Condition newCoverageCondition = analysis.findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)
-                    .orElseThrow(() -> new IllegalStateException("Could not find New Coverage Condition in analysis"));
-            String coverageValue = newCoverageCondition.getStatus().equals(QualityGate.EvaluationStatus.NO_VALUE) ? "0" : newCoverageCondition.getValue();
-
-
+            String coverageValue = analysis.findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)
+                .map(
+                    c ->  c.getStatus().equals(QualityGate.EvaluationStatus.NO_VALUE) ? "0" : c.getValue()
+                ).orElse("0")
+            ;
```
Comment (mc1arke) [inline src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabServerPullRequestDecorator.java:153]: Can this not be expressed without ternary operations as the following?
```suggestion
            String coverageValue = analysis.findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)
                .filter(c -> ! c.getStatus().equals(QualityGate.EvaluationStatus.NO_VALUE))
                .map(QualityGate.Condition::getValue)
                .orElse("0");
```

**Comment by mc1arke** [comment]
Superseded by the merge of #122

**Pull request closed without merging**